uro98/assignment3/FindAllWordsTest.java

```diff
@@ -0,0 +1,66 @@
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Test;
+
+public class FindAllWordsTest {
+
+  @Test
+  public void testFindAllWords() {
+    String[][] array = new String[][] {
+      {"A", "A", "R"},
+      {"T", "C", "D"}
+    };
+    
+    Grid grid = new Grid(2, 3);
+    grid.populate(array);
+    
+    Dictionary dictionary = new Dictionary();
+    dictionary.addWords(Arrays.asList("CAR", "CARD", "CART", "CAT"));
+    
+    Set<String> wordSet = FindAllWords.findAllWords(grid, dictionary);
+    
+    Set<String> expected = new HashSet<String>();
+    expected.add("CAR");
+    expected.add("CARD");
+    expected.add("CAT");
+    
+    assertEquals(expected, wordSet);
+    assertFalse(dictionary.getWordSet().equals(wordSet));
```

Comment:
Not necessary. If you want to test `Dictionary` (good!), it should have separate class with tests, so that one test class focuses on only one tested class.